subsys/bluetooth/mesh/Kconfig

```diff
@@ -37,7 +37,7 @@ config BT_MESH_GATT_CLIENT
 config BT_MESH_GATT_SERVER
 	bool
 	select BT_MESH_GATT
-	select BT_GATT_DYNAMIC_DB
+	depends on BT_GATT_DYNAMIC_DB
```

Comment:
If the Kconfig circular dependencies requires changes to everywhere, including mesh, then it should be fixed in a separate commit or maybe even a separate PR. Doing it together with the BASS changes is messy

Reply:
Then what’s your suggestion for this PR? Should I revert to using ´select´ (instead of ´depends on´) temporarily for ´BT_GATT_DYNAMIC_DB´ in ´BT_BAP_SCAN_DELEGATOR´ definition for this PR and then fix it in another PR?

Reply:
We can

1. Revert to using `select` and fix in another PR
2. Add a commit preceding this that fixes the dependency issues and then add the BASS commit on top of that
3. Create new PR fixing the dependency issues and then put this PR on top of that PR (creating 2 PRs). 

I don't have a strong preference on which direction to take, so will leave that up to you

Reply:
Ok, I prefer the first option :)
So, I'll update the PR accordingly...